crates/typst-library/src/meta/counter.rs

```diff
@@ -388,6 +388,29 @@ impl Counter {
         UpdateElem::new(self.0, update).pack()
     }
 
+    /// Calculate the difference between the real page number and the logical one.
+    ///
+    /// Difference between last counter change.
+    pub fn page_delta(&self, vt: &mut Vt, num: NonZeroUsize) -> SourceResult<usize> {
+        if !self.is_page() {
+            bail!(Span::detached(), "Only works for the page counter!");
+        }
+
+        let sequence = self.sequence(vt)?;
+        let offset = vt.introspector.query_up_to(num).len();
```

Comment:
Does not take a selector like @laurmaedje mentioned but this seemed to make the most sense.
I am also not 100% certain on correctness since this `Vec` will mostly consist of `DisplayElem`s and other stuff but
it seems to work.

Reply:
see my comment below